spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java

```diff
@@ -207,22 +207,20 @@ public Statistics estimateStatistics() {
       LOG.debug("using table metadata to estimate table statistics");
       long totalRecords = PropertyUtil.propertyAsLong(table.currentSnapshot().summary(),
           SnapshotSummary.TOTAL_RECORDS_PROP, Long.MAX_VALUE);
-      Schema projectedSchema = expectedSchema != null ? expectedSchema : table.schema();
```

Comment:
I don't think `expectedSchema` is ever null. Thus, I think we can just call `readSchema()` to get the `StructType`.